ddp-workspace/projects/ddp-sdk/src/lib/components/activityForm/activity.component.ts

```diff
@@ -190,28 +190,48 @@ export class ActivityComponent extends BaseActivityComponent implements OnInit,
             if (scroll) {
                 this.scrollToTop();
             }
-            // enable any validation errors to be visible
-            this.validationRequested = true;
-            this.sendSectionAnalytics();
-            this.currentSection.validate();
-            if (this.currentSection.valid) {
-                this.resetValidationState();
-                this.currentSectionIndex = nextIndex;
-                this.visitedSectionIndexes[nextIndex] = true;
-                this.saveLastVisitedSectionIndex(nextIndex);
-            }
+            this.submitAnnouncementService.announceSubmit(null);
+            // The announcement could make listener components busy, but not instantly
+            // introduce a wait before we check whether we busy or not
+            timer(100).pipe(
+                mergeMap(() =>
+                    this.isPageBusy.pipe(
+                        filter(pageIsBusy => !pageIsBusy),
+                        tap(() => {
+                            this.validationRequested = true;
+                            this.sendSectionAnalytics();
+                            this.currentSection.validate();
+                            if (this.currentSection.valid) {
+                                this.resetValidationState();
+                                this.visitedSectionIndexes[nextIndex] = true;
+                                this.saveLastVisitedSectionIndex(nextIndex);
+                                this.currentSectionIndex = nextIndex;
+                            }
+                        })
+                    )
+                ),
+                take(1)
+            ).subscribe();
         }
     }
 
     public decrementStep(scroll: boolean = true): void {
         const previousIndex = this.previousAvailableSectionIndex();
         if (previousIndex !== -1) {
-            // if we move forwards or backwards, let's reset our validation display
-            this.resetValidationState();
-            this.currentSectionIndex = previousIndex;
             if (scroll) {
                 this.scrollToTop();
             }
+            this.isPageBusy.pipe(startWith(true)).pipe(
+                delay(1),
+                filter(pageIsBusy => !pageIsBusy),
+                debounceTime(250),
+                tap(() => {
+                    // if we move forwards or backwards, let's reset our validation display
+                    this.resetValidationState();
+                    this.currentSectionIndex = previousIndex;
+                }),
+                take(1)
+            ).subscribe();
```

Comment:
Here to, we hold on with decrementing steps while the page is busy.
Note that we DO NOT do a submit announce! Seems to me that if going backwards we should not save address like going forwards.
Note that the mailing address does save copy of the fields automatically to temp address IF there is no real address. This happens anyways, so if some fields are entered for first time and hit the Previous button, when you come back the address fields entered should still be there.

Reply:
@alextanlan Could you please double-check my updates to isLoaded$ subject? I think in earlier version I had broken displaying next and previous section by setting by my use of `isLoaded$.next(false)` and don't think I fully understand how it fits in the bigger picture.

Reply:
I am in progress..